jmolecules-events/src/main/java/org/jmolecules/event/annotation/sourcing/SourcingEventHandler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jmolecules.event.annotation.sourcing;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Identifies a sourcing event handler, i.e. logic to process a {@link SourcingEvent}.
+ *
+ * @author Christian Stettler
+ * @author Henning Schwentner
+ * @author Stephan Pirnbaum
+ * @author Martin Schimak
+ * @author Oliver Drotbohm
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+@Documented
+public @interface SourcingEventHandler {
```

Comment:
As per my comment above, I don't believe that a "sourcing event" is a viable concept. However, a handler may have the intent to rebuild state based off a stream of events. In Axon, we have chosen the term "EventSourcingHandler" to mark a handler as being intended for "Event Sourcing", rather than Event Streaming.

Reply:
see #41